includes/datagrid-ref-confirm-action-and-validate-data.md

```diff
@@ -0,0 +1,187 @@
+                });
+                e.cancel = deferred.promise();
+            }
+        })
+    })
+
+##### Angular
+
+    <!--tab: app.component.ts-->
+    import { HttpClient, HttpClientModule, HttpParams } from "@angular/common/http";
+    import { confirm } from 'devextreme/ui/dialog';
+    // ...
+    export class AppComponent {
+        constructor(private httpClient: HttpClient) { /*...*/}
+        // ...
+        async ${{functionName}}(e) {
+            const isCanceled = async () => {
+                const dialogResult = await confirm("Are you sure?", "Confirm changes");
+                if (dialogResult) {
+                    let params = new HttpParams();
+                    for (let key in e.newData) {
+                        params = params.set(key, e.newData[key]);
+                    }
+                    const validationResult = await this.httpClient
+                        .get("https://url/to/your/validation/service", { params: params })
+                        .toPromise();
+                    if (validationResult.errorText) {
+                        console.log(validationResult.errorText);
+                        return true;
+                    } else {
+                        return false;
+                    } 
+                } else {
+                    return true;
+                }
+            }
+            e.cancel = await isCanceled();
+        }
+    }
+
+    <!-- tab: app.component.html -->
+    <dx-{widget-name} ... 
+        (${{apiMember}})="${{functionName}}($event)">
+    </dx-{widget-name}>
+
+    <!-- tab: app.module.ts -->
+    // ... 
+    import { Dx{WidgetName}Module } from 'devextreme-angular'; 
+    import { HttpClientModule } from "@angular/common/http";
+
+    @NgModule({
+        imports: [
+            // ...
+            Dx{WidgetName}Module,
+            HttpClientModule
+        ],
+        // ...
+    })
+
+##### Vue
+
+    <!-- tab: App.vue -->
+    <template>
+        <Dx{WidgetName} ...
+            @${{vueAttribute}}="${{functionName}}">
+        </Dx{WidgetName}>
+    </template>
+    <script>
+    import Dx{WidgetName}, { ... } from 'devextreme-vue/{widget-name}';
+    import dialog from 'devextreme/ui/dialog';
+    // ...
+
+    export default {
+        components: {
+            Dx{WidgetName},
+            // ...
+        },
+        // ...
+        methods: {
+            async ${{functionName}}(e) {
+                const isCanceled = async () => {
+                    const dialogResult = await dialog.confirm("Are you sure?", "Confirm changes");
+                    if (dialogResult) {
+                        let params = "?";
+                        for (let key in e.newData) {
+                            params += `${key}=${e.newData[key]}&`;
+                        }
+                        params = params.slice(0, -1);
+                        const validationResult = await fetch(`https://url/to/your/validation/service${params}`);
```

Comment:
why not pass params as body to fetch?
https://developer.mozilla.org/en-US/docs/Web/API/Fetch_API/Using_Fetch